homework1/linearInterpolation.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import pandas as pd
+from random import uniform
+
+class Interpolation(object):
+
+    def __init__(self, X, z):
+        if isinstance(X, list):
```

Comment:
More comments are needed in description on what is class doing, and what functions are doing, around specific operations and on what input is expected